terraform/environment/api/swagger.json

```diff
@@ -0,0 +1,159 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "OPG Metrics Rest API",
+    "version": "0.0.1"
+  },
+  "basePath": "/kinesis",
+  "schemes": ["https"],
+  "x-amazon-apigateway-request-validators": {
+    "Validate body": {
+      "validateRequestParameters": false,
+      "validateRequestBody": true
+    }
+  },
+  "paths": {
+    "/metrics": {
+      "put": {
+        "consumes": ["application/json"],
+        "produces": ["application/json"],
+        "parameters": [
+          {
+            "in": "body",
+            "name": "PutMetrics",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/PutMetrics"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
Wonder if putting metrics should be a 202 accepted?

Reply:
You are right, in this instance it is a fire and forget onto a seperate async process so this fits nicely. As well as that I am going to change the response so that it returns an appropriate message of the status of the request. Something like "Processing" as defined in the spec